Code/autopkglib/URLGetter.py

```diff
@@ -71,6 +71,15 @@ def produce_etag_headers(self, filename):
         headers = {}
         # If the download file already exists, add some headers to the request
         # so we don't retrieve the content if it hasn't changed
+        if self.env["external_metadata_path"]:
+            metadata_object = metadata.Metadata(self.env["external_metadata_path"])
+            etag = metadata_object.getmetadata(filename, self.xattr_etag)
+            last_modified = metadata_object.getmetadata(filename, self.xattr_last_modified)
+            if etag:
+                headers["If-None-Match"] = etag
+            if last_modified:
+                headers["If-Modified-Since"] = last_modified
+            return headers
         if os.path.exists(filename):
```

Comment:
I'm in favor of making this line an `elif` and keeping only a single `return headers` at the end of the function.

Reply:
Fixed with an elif